File: src/features/typing/data/java/longest_repeating_character_replacement.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int characterReplacement(String s, int k) {
        Map<Character, Integer> count = new HashMap<>();
        int maxCount = 0;
        int left = 0;
        int best = 0;
        for (int right = 0; right < s.length(); right++) {
            char c = s.charAt(right);
            count.put(c, count.getOrDefault(c, 0) + 1);
            if (count.get(c) > maxCount) {
                maxCount = count.get(c);
            }
            while (right - left + 1 > maxCount + k) {
                char leftChar = s.charAt(left);
                count.put(leftChar, count.get(leftChar) - 1);
                left++;
            }
            if (right - left + 1 > best) {
                best = right - left + 1;
            }
        }
        return best;
    }
}

File: src/features/typing/data/dart/longest_repeating_character_replacement.dart
class Solution {
  int characterReplacement(String s, int k) {
    final count = <String, int>{};
    var maxCount = 0;
    var left = 0;
    var best = 0;
    for (var right = 0; right < s.length; right++) {
      final c = s[right];
      count[c] = (count[c] ?? 0) + 1;
      if (count[c]! > maxCount) {
        maxCount = count[c]!;
      }
      while (right - left + 1 > maxCount + k) {
        final leftChar = s[left];
        count[leftChar] = count[leftChar]! - 1;
        left++;
      }
      if (right - left + 1 > best) {
        best = right - left + 1;
      }
    }
    return best;
  }
}
